### PR TITLE
Attributes contribute to message size restrictions

### PR DESCRIPTION
**Pull request by danielmarbach**
Replaces https://github.com/Particular/NServiceBus.AmazonSQS/pull/319
Fixes https://github.com/Particular/NServiceBus.AmazonSQS/issues/318

### CODE REVIEW

**Comment by danielmarbach** [comment]
This also need to extend into the batcher and properly cache the message size. So consider this as WIP

File: src/NServiceBus.AmazonSQS/PreparedMessage.cs [hunk 1]
```diff
@@ -1,18 +1,64 @@
...
         public Dictionary<string, MessageAttributeValue> MessageAttributes { get; } = new Dictionary<string, MessageAttributeValue>();
         public string MessageGroupId { get; set; }
         public string MessageDeduplicationId { get; set; }
+
+        public long CalculateSize()
+        {
+            long size = Body.Length;
+            foreach (var messageAttributeValue in MessageAttributes)
+            {
+                size += messageAttributeValue.Key.Length;
```
Comment (rafalkozik) [inline src/NServiceBus.AmazonSQS/PreparedMessage.cs]: From my experiments it looks that _DataType_ counts as well, sending this: 

`	var sendMessageBatchRequest = new SendMessageBatchRequest();

	sendMessageBatchRequest.Entries.Add(new SendMessageBatchRequestEntry(Guid.NewGuid().ToString(), new string('x', 1024 * 256))
	{
		MessageAttributes = new Dictionary<string, MessageAttributeValue>
		{
			["foo"] = new MessageAttributeValue { DataType = "String", StringValue = "abcdef" },
			["bar"] = new MessageAttributeValue { DataType = "String", StringValue = "xyz" }
		},
	});`

Results in _One or more errors occurred. (Batch requests cannot be longer than 262144 bytes. You have sent 262171 bytes.)_, which is 27 bytes above the limit.

Comment (danielmarbach) [reply src/NServiceBus.AmazonSQS/PreparedMessage.cs]: Interesting. I reread the spec again and saw now

> The data type has the same restrictions as the message body (for more information)

It seems in my first read-through I thought it only applies for custom data types. Will fix

**Comment by danielmarbach** [comment]
Replaced by https://github.com/Particular/NServiceBus.AmazonSQS/pull/329

**Pull request closed without merging**